aws/src/utils/safe_object_from_dict.py

```diff
@@ -0,0 +1,25 @@
+from typing import T
+
+
+def safe_object_from_dict(object_type: type[T], data: dict) -> T:
+    """
+    Safely creates an object of the given type from the given dictionary.
+    :param object_type: The type of object to create.
+    :param data: The dictionary containing the data to use for creating the object.
+    :return: An object of the given type, created from the given dictionary.
+    """
+
+    if data is None:
+        return None
+
+    for key in list(data.keys()):
+        if key not in object_type.__annotations__.keys():
+            print(f"WARNING: {key} not in {object_type.__name__}")
+            del data[key]
+
+    for key in object_type.__annotations__.keys():
+        if key not in data.keys():
+            print(f"WARNING: {key} from {object_type.__name__} not in data")
+            data[key] = None
+
+    return object_type(**data)
```

Comment:
This key removal & set to None logic will take care off the top-most level of an object.

So an object based on say "Tenure" class, will convert the children objects keys to expected types, but will it do the line15-23 schema-fix logic?
`householdMembers: list[HouseholdMember] | None`

Try adding a test, where householdMembers has an key that's not defined by the type. Let's see what happens.

You could traverse the whole object tree with a bit of type checking & recursion if it turns out the child object keys do not get the lines15-23 applied to them.

Reply:
The conversion of the inner types is handled by the `post_init` method on the Tenure data class:
```
def __post_init__(self):
        self.tenuredAsset = safe_object_from_dict(TenuredAsset, self.tenuredAsset)
        self.householdMembers = [safe_object_from_dict(HouseholdMember, member) for member in self.householdMembers]
```

Added a passing test for it as this is definitely worth verifying, thanks for the pointer.
See: https://github.com/LBHackney-IT/mtfh-scripts/pull/9/commits/cd6304497b3ddd131e5c10cadf79e0abe1fef2a3